ckanclient/__init__.py

```diff
@@ -0,0 +1,165 @@
+import logging
+
+import requests
+from frictionless_ckan_mapper import ckan_to_frictionless, frictionless_to_ckan
+
+from ckanclient.auth import CkanAuthApi
+from ckanclient.text import camel_to_snake
+from ckanclient.upload import push_data_to_blob_storage, verify_upload
+
+
+LOG_FORMAT = '%(asctime)s - %(levelname)s - %(message)s'
+logging.basicConfig(format=LOG_FORMAT, level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class CkanClientError(Exception):
+    pass
+
+
+class CkanClient:
+    def __init__(self, api_url, api_key, organization, dataset_id, lfs_url):
+        self.api_url = api_url
+        self.api_key = api_key
+        self.organization = organization
+        self.dataset_id = dataset_id
+        self.lfs_url = lfs_url
+        self.auth = CkanAuthApi(api_url, api_key, organization, dataset_id)
+
+    def action(
+        self,
+        name,
+        payload,
+        http_get=False,
+        transform_payload=None,
+        transform_response=None,
+    ):
+        """Gives you direct access to the CKAN Action API
+        (https://docs.ckan.org/en/2.8/api/).
+
+        Attributes:
+            name (str): The action name, e.g. site_read, package_show…
+            payload (dict): The payload being sent to CKAN. If a payload is
+                provided for a GET request, it will be converted to URL
+                parameters and each key will be converted to snake case.
+            http_get (bool): Optional, if `True` will make `GET` request,
+                otherwise `POST`.
+            transform_payload (func): Function to mutate the `payload` before
+                making the request (useful to convert to and from CKAN and
+                Frictionless formats).
+            transform_response (func): Function to mutate the response data
+                before returning it (useful to convert to and from CKAN and
+                Frictionless formats).
+
+        Returns:
+            dict: the API response converted from JSON to a Python dictionary.
+        """
+        url = f'{self.api_url}/api/3/action/{name}'
```

Comment:
Was there a reason not to use http://github.com/ckan/ckanapi - then we could just reuse that.